### PR TITLE
CI/workflows

### PR DESCRIPTION
**Pull request by apollo-bobby**
## Todo
- [x] Enforce conventional commits on the CI
- [x] Fix bad formatting in sailr messages on commit
- [x] Improve/add QoL fixes to pre-commit hook
- [x] Run `git init` at the end of `install-git-hooks.sh`

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "https://github.com/apollodao/cw-asset",
+    "https://github.com/Wasmswap/wasmswap-contracts",
+    "https://github.com/DA0-DA0/dao-contracts",
+    "https://github.com/astroport-fi/astroport-core",
+]
+
+[licenses]
+unlicensed = "allow"           # TODO: update this
+allow-osi-fsf-free = "neither"
+copyleft = "allow"             # TODO: update this
```
Comment (apollo-sturdy) [inline deny.toml:35]: We should be able to change this now that we removed astroport as a dependency.

File: install-git-hooks.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+cp .pre-commit.sh .git/hooks/pre-commit
+chmod +x .git/hooks/pre-commit
+
+## Commit-msg
+if stat .git/hooks/commit-msg 2&>/dev/null; then
+    echo "========================================"
+    echo "=== Installing sailr commit-msg hook ==="
+    echo "========================================\n"
+    # sailr requires has `jq` to be installed
+    curl -o- https://raw.githubusercontent.com/craicoverflow/sailr/master/scripts/install.sh | bash
```
Comment (apollo-sturdy) [inline install-git-hooks.sh:23]: Is there a way to do this where we don't trust the maintainers of this repo? They could theoretically replace this with a malicious script right?

**Review by apollo-sturdy — APPROVED** [review]
Approving but see comments

File: install-git-hooks.sh (hunk 2 above)
Comment (apollo-sturdy) [reply install-git-hooks.sh:23]: I forked the repo so we should be good.

File: deny.toml (hunk 1 above)
Comment (apollo-sturdy) [reply deny.toml:35]: Fixed

File: .github/workflows/cc.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
+name: Conventional commit check
+on:
+  push:
+    branches:
+      - master
+
+# This workflow denies a push to master if all pushed commits are not following
+# the conventional commits standard.
```
[the remark this answers is not in the document]
Comment (apollo-sturdy) [reply .github/workflows/cc.yml:9]: Changed them to be on pull_request for now.

[the remark this answers is not in the document]
Comment (apollo-bobby) [reply .github/workflows/cc.yml:9]: > Also does merging a PR count as a push? Ideally we should never be directly pushing to master.

Yes, merging a PR always counts as a push. We can use branch protection rules to disable direct pushing to `master`.

I think I had a reason for doing it this way but I can't remember it now.

File: install-git-hooks.sh (hunk 2 above)
Comment (apollo-bobby) [reply install-git-hooks.sh]: I was planning on forking the repo anyway because the text formatting is broken (at least on my machine).

**Pull request merged**